components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/StreamingListener.java

```diff
@@ -0,0 +1,215 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.inbound.endpoint.protocol.nats;
+
+import io.nats.streaming.StreamingConnection;
+import io.nats.streaming.Subscription;
+import io.nats.streaming.Options;
+import io.nats.streaming.StreamingConnectionFactory;
+import io.nats.streaming.SubscriptionOptions;
+import io.nats.streaming.Message;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+import org.wso2.carbon.inbound.endpoint.protocol.nats.management.NatsEndpointManager;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Properties;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Streaming listener class which uses NATS streaming connection to receive messages.
+ */
+public class StreamingListener implements NatsMessageListener {
+
+    private static final Log log = LogFactory.getLog(StreamingListener.class.getName());
+    private String subject;
+    private NatsInjectHandler injectHandler;
+    private Properties natsProperties;
+    private StreamingConnection connection;
+    private Subscription subscription;
+
+    public StreamingListener(String subject, NatsInjectHandler injectHandler, Properties natsProperties) {
+        this.subject = subject;
+        this.injectHandler = injectHandler;
+        this.natsProperties = natsProperties;
+    }
+
+    /**
+     * Create the connection to the NATS Streaming server.
+     *
+     * @return boolean value whether connection is created.
+     */
+    @Override public boolean createConnection() throws IOException, InterruptedException {
+        if (connection == null) {
+            String natsStreamingUrl = natsProperties.getProperty(NatsConstants.NATS_STREAMING_URL);
+            String natsStreamingClientId = natsProperties.getProperty(NatsConstants.NATS_STREAMING_CLIENT_ID);
+            String natsStreamingClusterId = natsProperties.getProperty(NatsConstants.NATS_STREAMING_CLUSTER_ID);
+            String connectWait = natsProperties.getProperty(NatsConstants.NATS_STREAMING_CONNECT_WAIT);
+            String discoverPrefix = natsProperties.getProperty(NatsConstants.NATS_STREAMING_DISCOVER_PREFIX);
+            String maxPingsOut = natsProperties.getProperty(NatsConstants.NATS_STREAMING_MAX_PINGS_OUT);
+            String pingInterval = natsProperties.getProperty(NatsConstants.NATS_STREAMING_PING_INTERVAL);
+            String traceConnection = natsProperties.getProperty(NatsConstants.NATS_STREAMING_TRACE_CONNECTION);
+
+            Options.Builder builder = new Options.Builder().natsUrl(
```

Comment:
Isn't there SSL support for NATS streaming?

Reply:
No there isnt but we can configure a Core NATS connection with SSL and use it with the Streaming connection